src/recipe.h

```diff
@@ -50,6 +51,26 @@ struct recipe_proficiency {
     void deserialize( JsonIn &jsin );
 };
 
+struct book_recipe_data {
+    int skill_req = -1;
+    bool use_alt_name = false;
+    std::string alt_name = "";
+    bool hidden = false;
+
+    book_recipe_data( JsonArray arr ) {
+        int arr_size = arr.size();
+        if( arr_size > 1 ) {
+            skill_req = arr.get_int( 1 );
+        }
+        if( arr_size > 2 ) {
+            std::string name = arr.get_string( 2 );
```

Comment:
You should use `class translation` here for translatable strings.

Reply:
I hope I used that correctly.